### ISSUE TITLE
make fails with mongoose

### ISSUE THREAD

**Issue by skipper85** [whoshuu/cpr#387]
Hi,  
I would like to try out CPR but was not able to build.  
I received the next error:  
```
[  0%] Building C object opt/CMakeFiles/mongoose.dir/mongoose/mongoose.c.o
/home/gerard/Downloads/cpr/opt/mongoose/mongoose.c: In function ‘mg_copy_listeners’:
/home/gerard/Downloads/cpr/opt/mongoose/mongoose.c:5110:21: error: dereferencing pointer to incomplete type ‘SSL_CTX’ {aka ‘struct ssl_ctx_st’}
 5110 |         tmp->ssl_ctx->references++;
      |                     ^~
make[2]: *** [opt/CMakeFiles/mongoose.dir/build.make:63: opt/CMakeFiles/mongoose.dir/mongoose/mongoose.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:96: opt/CMakeFiles/mongoose.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

I am using Ubuntu 19:10 with:  
* gcc 9.2.1
* libcurl4-gnutls-dev
* libssl-dev
* cmake version 3.13.4
What do I miss to build the project?  

Thank you in advance for your help

**Comment by COM8** [comment]
This looks to me like an issue in the way you include CPR in your CMake file.
Could you please provide a minimal example project?

**Comment by COM8** [comment]
Ping @kingkili.

**Comment by COM8** [comment]
OK. I can confirm this issue.
It's the result of an outdated `mongoose` version.
This will be fixed by #364

**Comment by skipper85** [comment]
Hi COM8,

Thanks for the update.
I was not using it yet in a project, just building with CMakeLists.txt from this repo.
